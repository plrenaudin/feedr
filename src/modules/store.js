import { observable, action, toJS, configure } from "mobx";
import db, { EMPTY_RECORD } from "./db";
import { shortFormatDate } from "./formatter";
import addDays from "date-fns/add_days";

configure({ enforceActions: true });

export default class Store {
  day = observable({ ...EMPTY_RECORD, date: shortFormatDate() });
  constructor() {
    this.load(shortFormatDate());
  }

  set = action(({ date = shortFormatDate(), morning = [], noon = [], evening = [] }) => {
    this.day.date = date;
    this.day.morning = morning;
    this.day.noon = noon;
    this.day.evening = evening;
  });

  addItem = action((meal, data) => {
    this.day[meal].push(data);
    this.save();
  });

  removeItem = action((meal, index) => {
    this.day[meal].splice(index, 1);
    this.save();
  });

  nextDay() {
    this.load(shortFormatDate(addDays(this.day.date, 1)));
  }

  previousDay() {
    this.load(shortFormatDate(addDays(this.day.date, -1)));
  }

  load = action(date => {
    return db.get(date).then(data => {
      this.set(data || { date });
    });
  });

  save() {
    return db.set(this.day.date, toJS(this.day));
  }
}
