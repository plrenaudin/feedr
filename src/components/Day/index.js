import Meal from "../Meal";
import { inject, observer } from "inferno-mobx";
import { longFormatDate } from "../../modules/formatter";
import { linkEvent } from "inferno";

const editNotes = (store, event) => {
  store.editNotes(event.target.value);
};
const Day = ({ store }) => {
  const notes = store.day.notes;
  return (
    <section class="day">
      <h2>{longFormatDate(store.day.date)}</h2>
      <Meal meal="morning" />
      <Meal meal="noon" />
      <Meal meal="snack" />
      <Meal meal="evening" />
      <h3>
        <span class="fa-layers fa-fw">
          <i class="fas fa-circle" data-fa-transform="grow-6" />
          <i class="fas fa-edit" data-fa-transform="shrink-4" />
        </span>
        Notes
      </h3>
      <textarea
        onChange={linkEvent(store, editNotes)}
        ref={input => input && (input.value = notes)}
      />
    </section>
  );
};

export default inject("store")(observer(Day));
