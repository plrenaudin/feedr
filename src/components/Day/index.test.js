import { render } from "inferno";
import Day from ".";
import store from "../../../test/fixture/store";
import { Provider } from "inferno-mobx";

it("Matches the snapshot", () => {
  const div = document.createElement("div");
  const rendered = render(
    <Provider store={store}>
      <Day />
    </Provider>,
    div
  );
  expect(rendered.props.children.dom).toMatchSnapshot();
});
