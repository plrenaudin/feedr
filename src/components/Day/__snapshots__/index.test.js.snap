// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<section
  class="day"
>
  <h2>
    Thursday 7th June
  </h2>
  <div
    class="meal"
  >
    <h3>
      <span
        class="fa-layers fa-fw"
      >
        <i
          class="fas fa-circle"
          data-fa-transform="grow-6"
        />
        <i
          class="fas fa-utensils"
          data-fa-transform="shrink-2"
        />
      </span>
      Breakfast
    </h3>
    <div
      class="mealContent"
    >
      <ul>
        <li>
          <span>
            morning Meal 1
          </span>
          <span
            class="actions"
          >
            <div
              class="food-type-picker"
            >
              <a
                role="button"
              >
                <i
                  class="fas fa-tag"
                />
              </a>
            </div>
            <a
              role="button"
            >
              <i
                class="far fa-trash-alt"
              />
            </a>
          </span>
        </li>
      </ul>
      <input
        placeholder="What else...?"
        type="text"
      />
    </div>
  </div>
  <div
    class="meal"
  >
    <h3>
      <span
        class="fa-layers fa-fw"
      >
        <i
          class="fas fa-circle"
          data-fa-transform="grow-6"
        />
        <i
          class="fas fa-utensils"
          data-fa-transform="shrink-2"
        />
      </span>
      Lunch
    </h3>
    <div
      class="mealContent"
    >
      <ul />
      <input
        placeholder="What did you eat?"
        type="text"
      />
    </div>
  </div>
  <div
    class="meal"
  >
    <h3>
      <span
        class="fa-layers fa-fw"
      >
        <i
          class="fas fa-circle"
          data-fa-transform="grow-6"
        />
        <i
          class="fas fa-utensils"
          data-fa-transform="shrink-2"
        />
      </span>
      Diner
    </h3>
    <div
      class="mealContent"
    >
      <ul>
        <li>
          <span>
            evening Meal 1
          </span>
          <span
            class="actions"
          >
            <div
              class="food-type-picker"
            >
              <a
                role="button"
              >
                <i
                  class="fas fa-tag"
                />
              </a>
            </div>
            <a
              role="button"
            >
              <i
                class="far fa-trash-alt"
              />
            </a>
          </span>
        </li>
        <li>
          <span>
            evening Meal 2
          </span>
          <span
            class="actions"
          >
            <div
              class="food-type-picker"
            >
              <a
                role="button"
              >
                <i
                  class="fas fa-tag"
                />
              </a>
            </div>
            <a
              role="button"
            >
              <i
                class="far fa-trash-alt"
              />
            </a>
          </span>
        </li>
      </ul>
      <input
        placeholder="What else...?"
        type="text"
      />
    </div>
  </div>
  <h3>
    <span
      class="fa-layers fa-fw"
    >
      <i
        class="fas fa-circle"
        data-fa-transform="grow-6"
      />
      <i
        class="fas fa-edit"
        data-fa-transform="shrink-4"
      />
    </span>
    Notes
  </h3>
  <textarea />
</section>
`;
